HeliosLib/Makefile

```diff
@@ -108,25 +118,45 @@ tests: $(TESTS)
 
 # force sub-build of wasm
 wasm: FORCE
-	env WASM=1 $(MAKE)
+	@if command -v em++; then \
+		echo "WebAssembly compiler found, building WASM..."; \
+		env ARCH=wasm WASM=1 $(MAKE) $(FINAL_WASM_TARGET) $(FINAL_LIB_TARGET); \
+	else \
+		echo "WebAssembly compiler (em++) not found, skipping WASM build."; \
+		mkdir -p $(OUTPUT_DIR); \
+		touch $(FINAL_WASM_TARGET); \
```

Comment:
why are these here? If it's skipping the wasm build in this section why mkdir and touch files?